app/js/components/maintenance/maintenanceCategorySelect/maintenanceCategorySelect.directive.js

```diff
@@ -50,8 +54,17 @@ function maintenanceCategorySelectCtrl($rootScope,
         return option.tag;
     }
 
+    function setUrl() {
+        if ($scope.selected.id !== 'all') {
+            $location.search('maintenanceCategory', $scope.selected.id);
+        } else if ($location.search().maintenanceCategory) {
+            delete $location.search().maintenanceCategory;
+        }
+    }
```

Comment:
not entirely sure why but the URL is not being updated. I'm thinking that it has something to do with maintenance.controller.js updating the URL and overwriting anything that this component is doing to the URL.